src/VisualStudio/Core/Def/NavigateTo/RoslynCodeSearchResult.cs

```diff
@@ -37,7 +38,8 @@ public RoslynCodeSearchResult(
                   patternMatches,
                   location,
                   perProviderItemPriority: perProviderItemPriority,
-                  language: language)
+                  language: language,
+                  isActiveDocument: isActiveDocument)
```

Comment:
what impact does htis end up having?

Reply:
When you fixed #77051, you mentioned that we should also ensure that when the user has "Current Document" scope set in AIOS, we should prioritize results that are literally from the active document.

On the AIOS side, it's a little expensive to extract that info at ranking time. I thought it might make sense for the provider to set it instead, since you already check it for calculating priority.

So the idea is that I added `CodeSearchResult.IsActiveDocument` for providers to optionally set, and when we sort results to display to the user, any results with that set to `true` rise to the top when the user has "Current Document" scope set.

Reply:
wfm.